application/controllers/rest/V1/SurveyController.php

```diff
@@ -0,0 +1,219 @@
+     *          )
+     *      ),
+     *      @OA\Response(
+     *          response="400",
+     *          description="Bad request"
+     *      ),
+     *      @OA\Response(
+     *          response="401",
+     *          description="Unauthorized"
+     *      ),
+     *      @OA\Response(
+     *          response="404",
+     *          description="Not found"
+     *      )
+     * )
+     *
+     * @return void
+     */
+    public function actionIndexPost()
+    {
+        $request = Yii::app()->request;
+        $data = $request->getRestParams();
+        $commandRequest = new Request(array(
+            'sessionKey' => $this->getAuthToken(),
+            'surveyID' => isset($data['surveyID']) ? $data['surveyID'] : '',
+            'surveyTitle' => isset($data['surveyTitle']) ? $data['surveyTitle'] : '',
+            'surveyLanguage' => isset($data['surveyLanguage']) ? $data['surveyLanguage'] : '',
+            'format' => isset($data['format']) ? $data['format'] : null
+        ));
+        $commandResponse = (new SurveyAdd())
+            ->run($commandRequest);
+
+        $this->renderCommandResponse($commandResponse);
+    }
+
+    /**
+     * Handle get request
+     *
+     * @OA\Get(
+     *      path="/rest/v1/survey/{id}",
+     *      security={{"bearerAuth":{}}},
+     *      summary="Get survey by id",
+     *      description="Get survey by id",
+     *      tags={"Survey"},
+     *      @OA\Parameter(
+     *          description="Survey id",
+     *          in="path",
+     *          name="id",
+     *          required=true,
+     *          @OA\Schema(type="string")
+     *      ),
+     *      @OA\Parameter(
+     *          parameter="surveySettings",
+     *          name="surveySettings[]",
+     *          allowReserved=true,
+     *          in="query",
+     *          description="Survey settings",
+     *          required=false,
+     *          @OA\Schema(
+     *              type="array",
+     *              @OA\Items(type="string")
+     *          )
+     *      ),
+     *      @OA\Response(
+     *          response=200,
+     *          description="Success"
+     *      ),
+     *      @OA\Response(
+     *          response="400",
+     *          description="Bad request"
+     *      ),
+     *      @OA\Response(
+     *          response="401",
+     *          description="Unauthorized"
+     *      ),
+     *      @OA\Response(
+     *          response="404",
+     *          description="Not found"
+     *      )
+     * )
+     *
+     * @param string $id
+     * @return void
+     */
+    public function actionIndexGet(string $id)
+    {
+        $request = Yii::app()->request;
+        $requestData = array(
+            'sessionKey' => $this->getAuthToken(),
+            'surveyID' => $id,
+            'surveySettings' => $request->getParam('surveySettings')
+        );
+        $commandRequest = new Request($requestData);
+        $commandResponse = (new SurveyPropertiesGet())
```

Comment:
@fosteruk  I think there's some code repetition here that can be removed by either reflection or mapping request name to command object name, like

```php
[
  ['GET', 'survey'] => 'SurveyPropertiesGet'
]
```
Or something like that. :)